Fix a few places we almost properly check for null

Redth opened the pull request:
### Description of Change

There were a few places we weren't fully checking for null, or rather we checked for null for some operations but not others directly around or after.  This should cause a few less null ref exceptions.

Fixes #10517

Redth committed:
Fix a few places we almost properly check for null

campersau commented on src/Controls/src/Core/FormattedString.cs:
```diff
@@ -41,9 +41,9 @@ void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
-					bo.Parent = null;
 					if (bo != null)
 					{
+						bo.Parent = null;
```
Maybe also fix line 59? (That's why I personally prefer the `if (a is X x)` style instead of `as`.)

Redth committed:
Another one

Redth committed:
Remove unnecessary (and unused) trim call

Above (~line 108) when part is created/assigned, .Trim() is already run.  It's not reassigned from that, so no need to trim again, and also the result was unused as it was anyway.
